### ISSUE TITLE
OpenShift fails to start when NetworkPlugin=cni and no available CNI configs

### ISSUE THREAD

**Issue by djosborne** [openshift/origin#11907]
Description: OpenShift fails to start with NetworkPlugin=cni, when there is no configurations in the CNI config dir. 

Kubernetes has made moves to support CNI plugin installation using hosted installs on top of a running Kubernetes cluster. This means a cluster need only be configured with `--network-plugin=cni` and a `--network-plugin-dir` to be CNI ready, assuming a hosted install will then tackle the remaining install steps (installing the CNI plugin binaries, adding a valid cni.conf, etc.).

I'm finding that a Hosted install for CNI is impossible as the OpenShift cluster won't start when the network-plugin is set. This means hosted installs are broken for OpenShift, and all CNI plugins must be manually installed.

I'm new to OpenShift, so looking for any advice on how to proceed. I'm thinking its possible that Kubernetes has the same "bug", but Kubernetes ships with a slew of default CNI configurations so it is never encountered. Its also possible that the bug is intentional, to prevent users from starting CNI cluster  if there are no configured CNI plugins.

##### Version
I'm using the OpenShift Vagrant demo
```
$ oc version
oc v1.3.0
kubernetes v1.3.0+52492b4
features: Basic-Auth
```

##### Steps To Reproduce
1. Configure Openshift node with: 
   ```
   networkConfig:
     networkPluginName: cni
   ```

2. Openshift fails to start with error: 
   ```
   584 server.go:293] Failed running kubelet: failed to create kubelet: Network plugin "cni" not found.
   584 node.go:315] failed to create kubelet: Network plugin "cni" not found.
   ```

3. Create a garbage CNI configuration file in the default CNI configuration directory:
   ```
   cat << EOF | sudo tee /usr/libexec/kubernetes/kubelet-plugins/net/exec/fake.conf
   {}
   EOF
   ```
4. Openshift now starts succesfully.

**Comment by djosborne** [comment]
I've learned that this is a known issue with Kubernetes v1.3 which has been fixed in v1.4. Presumably openshift v1.4 will pick this up. Ok to close

**Comment by knobunc** [comment]
Thanks @djosborne
